### PR TITLE
DOC: low_memory in read_csv

### LINKED ISSUE #5888: API/DOC: status of low_memory kwarg of read_csv/table
I am getting the following warning:
```
/usr/local/lib/python2.7/dist-packages/pandas-0.13.0rc1_78_g142ca62-py2.7-linux-x86_64.egg/pandas/io/parsers.py:1050: 
DtypeWarning: Columns (6) have mixed types. Specify dtype option on import or set low_memory=False.
  data = self._reader.read(nrows)
```

but I can find no documentation for ```low_memory```

### PR DESCRIPTION
**Pull request by chris-b1**
- [x] closes #5888,  xref #12686
- [x] passes `git diff upstream/master | flake8 --diff`

### CODE REVIEW

File: doc/source/io.rst [hunk 1]
```diff
@@ -169,6 +169,13 @@ skipfooter : int, default ``0``
   Number of lines at bottom of file to skip (unsupported with engine='c').
 nrows : int, default ``None``
   Number of rows of file to read. Useful for reading pieces of large files.
+low_memory : boolean, default ``True``
+  Internally process the file in chunks, resulting in lower memory use
+  while parsing, but possibly mixed type inference.  To ensure no mixed
+  types either set False, or specify the type with the `dtype` parameter.
```
Comment (jreback) [inline doc/source/io.rst]: double back-ticks on False

Comment (jreback) [inline doc/source/io.rst]: dtype as well

**Comment by chris-b1** [comment]
@jreback - updated

**Comment by jreback** [comment]
thanks!

File: doc/source/io.rst [hunk 2]
```diff
@@ -169,6 +169,13 @@ skipfooter : int, default ``0``
   Number of lines at bottom of file to skip (unsupported with engine='c').
 nrows : int, default ``None``
   Number of rows of file to read. Useful for reading pieces of large files.
+low_memory : boolean, default ``True``
+  Internally process the file in chunks, resulting in lower memory use
+  while parsing, but possibly mixed type inference.  To ensure no mixed
```
Comment (chris-b1) [inline doc/source/io.rst]: It is deterministic - types are consistently inferred based on what's in the data.  That said, the internal chunksize is not a fixed number of rows, but instead bytes, so whether you can a mixed dtype warning or not can feel a bit random.

```python
from io import StringIO

# just ints - no warning
pd.read_csv(StringIO(
    '\n'.join(str(x) for x in range(10000)))
).dtypes
Out[24]: 
0    int64
dtype: object

# mixed dtype - but fits into a single chunk, so no warning and all parsed as strings
pd.read_csv(StringIO(
    '\n'.join([str(x) for x in range(10000)] + ['a string']))
).dtypes
Out[26]: 
0    object
dtype: object

# mixed dtype - doesn't fit into a single chunk
pd.read_csv(StringIO(
    '\n'.join([str(x) for x in range(1000000)] + ['a string']))
).dtypes
DtypeWarning: Columns (0) have mixed types. 
Specify dtype option on import or set low_memory=False.

Out[27]: 
0    object
dtype: object
```

Comment (jondo) [inline doc/source/io.rst]: I see. Thank you for the explanation and the example!

Comment (jondo) [inline doc/source/io.rst]: To complete the example, I see:
```python
df=pd.read_csv(StringIO('\n'.join([str(x) for x in range(1000000)] + ['a string'])))
DtypeWarning: Columns (0) have mixed types. Specify dtype option on import or set low_memory=False.

type(df.loc[524287,'0'])
Out[50]: int

type(df.loc[524288,'0'])
Out[51]: str
```
The first part of the csv data was seen as only int, so converted to int,
the second part also had a string, so all entries were kept as string.

**Pull request closed without merging**